File: Algorithms/1356.根据数字二进制下1的数目排序/sortByBits.java
// 方法1：暴力法
// 17 ms
class Solution {
    public int[] sortByBits(int[] arr) {
        List<Item> items = new ArrayList<>();
		for (int x : arr)
			items.add(new Item(x, Integer.bitCount(x)));
		items.sort(Comparator.comparing(Item::getBitCount).thenComparing(Item::getNum));
		return items.stream().mapToInt(Item::getNum).toArray();
    }
}

class Item {
	int num;
	int bitCount;

	Item(int num, int bitCount) {
		this.num = num;
		this.bitCount = bitCount;
	}

	public int getNum() {
		return num;
	}

	public int getBitCount() {
		return bitCount;
	}

}

File: Algorithms/1356.根据数字二进制下1的数目排序/sortByBits.py
class Solution:
    def sortByBits(self, arr: List[int]) -> List[int]:
        a = [(bit_count(x), x) for x in arr]
        a.sort()
        return [x for c, x in a]


def bit_count(x):
    count = 0
    while x:
        count += x & 1
        x >>= 1
    return count
